server/src/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -4788,7 +4788,7 @@ protected void validateNtwkOffDetails(final Map<Detail, String> details, final M
             // Now apply pagination
             final List<? extends NetworkOffering> wPagination = StringUtils.applyPagination(supportedOfferings, cmd.getStartIndex(), cmd.getPageSizeVal());
             if (wPagination != null) {
-                final Pair<List<? extends NetworkOffering>, Integer> listWPagination = new Pair<List<? extends NetworkOffering>, Integer>(wPagination, offerings.size());
+                final Pair<List<? extends NetworkOffering>, Integer> listWPagination = new Pair<List<? extends NetworkOffering>, Integer>(wPagination, supportedOfferings.size());
```

Comment:
Additional review and testing requested, if we need to change this line, then 4798 may have similar code as well.

Reply:
Line 4798 is correct. The supportedOfferings list is build within the ` if (parseOfferings)` code block. In the else case we want to return the offerings list and the size of the offerings list (supportedOfferings is also not defined in that case).

Reply:
@sgoeminn why not use `wPagination.size()` ?

Reply:
@rhtyd we always want to return the number of total found offerings and not only the number of offerings in the page that is returned. This is the same behavior as all the other list cmds.

Reply:
@sgoeminn thanks.